### PR TITLE
Users/mkonjikovac/advanced azure function example

### PR DESCRIPTION
**Pull request by mkonjikovac**
Basic and advanced Azure Function examples for working with Azure DevOps async checks.
Code changes have been built on top of the earlier examples and projects have been updated to use .NET 6 framework.

### CODE REVIEW

File: ServerTaskHelper/AzureFunctionAdvancedHandler/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+4. Checks if the ticket is in the `Completed` state
+5. Sends a status update with the result of the check
+6. If the ticket isn't in the Completed state, it reschedules another evaluation in 1 minute
+7. Once the ticket is in the correct state, it sends a positive decision to Azure Pipelines
+
+### In order for this example to work, following instructions should be followed for the setup:
+
+1. Deploy and configure AzureFunctionAdvancedHandler and AzureFunctionAdvancedServiceBusTrigger on azure portal.
+   Both of these functions have to be deployed in order for this example to work because they are dependent of each other.
+   AzureFunctionAdvancedHandler is entry point for the check process, where as AzureFunctionAdvancedServiceBusTrigger function
```
Comment (bsmid) [inline ServerTaskHelper/AzureFunctionAdvancedHandler/README.md:20]: where as -> whereas

File: ServerTaskHelper/AzureFunctionBasicHandler/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+1. Confirms the receipt of the check payload
+2. Sends a status update to Azure Pipelines that the check started
+3. Uses {AuthToken} to make a callback into Azure Pipelines to retrieve the pipeline run's Timeline entry
+4. Checks if the Timeline contains a CmdLine task
+5. Sends a status update with the result of the search
+6. Sends a check decision to Azure Pipelines
+
+### In order for this example to work, following instructions should be followed for the setup:
+1. Deploy azure function AzureFunctionBasicHandler to azure.
```
Comment (bsmid) [inline ServerTaskHelper/AzureFunctionBasicHandler/README.md:16]: azure function -> Azure Function? Maybe check other places in docs so we use consistent casing.

File: ServerTaskHelper/AzureFunctionBasicSample/CheckLogger.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+            Directory.CreateDirectory(pagesFolder);
+
+            var logFileName = Path.Combine(pagesFolder, $"taskLog.log");
+            var pageData = new FileStream(logFileName, FileMode.CreateNew);
+            var pageWriter = new StreamWriter(pageData, Encoding.UTF8);
+
+            SendTaskLogFeeds(httpClient, authToken, planUrl, projectId, hubName, planId, jobId, timelineId, message);
+            pageWriter.WriteLine(message);
+            pageWriter.Flush();
+            pageData.Flush();
```
Comment (bsmid) [inline ServerTaskHelper/AzureFunctionBasicSample/CheckLogger.cs:44]: Have to `flush` both? Flushing the `pageWriter` should be enough.

Comment (mkonjikovac) [reply ServerTaskHelper/AzureFunctionBasicSample/CheckLogger.cs:44]: You're probably right, though as agreed I am removing this project from the solution as it is in a way a duplicate of AzureFunctionBasicHandler and would just cause confusion.

File: ServerTaskHelper/AzureFunctionAdvancedHandler/MyTaskExecutionHandler.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+using DistributedTask.ServerTask.Remote.Common.TaskProgress;
+using DistributedTask.ServerTask.Remote.Common.WorkItemProgress;
+using DistributedTask.ServerTask.Remote.Common.WorkItemProgress.Exceptions;
+using Microsoft.Extensions.Logging;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Common;
+
+namespace AzureFunctionAdvancedHandler
+{
+    internal class MyTaskExecutionHandler
```
[the remark this answers is not in the document]
Comment (mkonjikovac) [reply ServerTaskHelper/AzureFunctionAdvancedHandler/MyTaskExecutionHandler.cs]: You're right, updated the names to: WorkItemStatusHandler.cs (for advanced Azure Function) and WorkItemStatusHandler.cs (for the basic one).

**Pull request merged**